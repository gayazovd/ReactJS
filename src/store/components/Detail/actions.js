export const SHOW_FILM_DETAIL = 'SHOW_FILM_DETAIL';

export const getFilm = movie => {
  return {
    type: SHOW_FILM_DETAIL,
    movie
  };
};
