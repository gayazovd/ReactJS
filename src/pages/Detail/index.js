import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import FilmDetail from './components/Movie';
import MovieList from '../../components/MovieList';
import InformationPanel from '../../components/InformationPanel';
import Footer from '../../components/Footer';
import Loading from '../../components/Loading';

import { getDetailFilm } from '../../store/components/Default/actions/actionCreators';

class Detail extends PureComponent {
    constructor(props) {
        super(props);
        const { id } = this.props.match.params;
        this.props.getDetailFilm(`/movies/${id}`);
    }

    componentDidMount() {
        window.scrollTo(0, 0);
    }

    componentDidUpdate(prevProps) {
        window.scrollTo(0, 0);
        const { id } = this.props.match.params;
        if (id != prevProps.match.params.id) {
            this.props.getDetailFilm(`/movies/${id}`);
        }
    }

    render() {
        const { movies, loading } = this.props;
        return (
            <>
                <FilmDetail />
                <InformationPanel />
                <MovieList data={movies} />
                <Footer />
                <Loading loading={loading} />
            </>
        );
    }
}

Detail.propTypes = {
    movies: PropTypes.arrayOf(PropTypes.object).isRequired
};

const mapStateToProps = state => {
    return {
        movies: state.loadingMovies.movies,
        loading: state.filmPreview.loading
    };
};

const mapDispatchToProps = {
    getDetailFilm
};
export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(Detail)
);
