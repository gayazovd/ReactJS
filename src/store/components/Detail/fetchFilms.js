import { getFilm } from './actions';
import { network } from '../../../services/network';

const getDataFromServer = url => dispatch => {
  network(url).then(data => {
    const { data: movie } = data;
    dispatch(getFilm(movie));
  });
};

export default getDataFromServer;
