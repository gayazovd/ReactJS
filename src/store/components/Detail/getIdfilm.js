import { getId } from './actions';

export const getIdByFilm = id => dispatch => dispatch(getId(id));
