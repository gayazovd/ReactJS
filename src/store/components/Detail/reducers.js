import { SHOW_FILM_DETAIL } from './actions';
import { GET_ID_FILM } from './actions';

const movie = (state = {}, action) => {
  switch (action.type) {
    case SHOW_FILM_DETAIL:
      return action.movie;
    case GET_ID_FILM:
      return { ...state, id: action.id };
    default:
      return state;
  }
};

export default movie;
