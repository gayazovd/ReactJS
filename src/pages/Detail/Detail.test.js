import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Detail from '../../pages/Detail';

configure({ adapter: new Adapter() });

describe('<Detail/> component testing', () => {
  const detail = shallow(<Detail />);
  it('component <Detail /> snapshot', () => {
    expect(detail).toMatchSnapshot();
  });
});
