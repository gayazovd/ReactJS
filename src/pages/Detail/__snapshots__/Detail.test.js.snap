// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail/> component testing component <Detail /> snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
